pkg/controller/mysqlpvc/mysqlpvc_controller.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2018 Platform9, Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mysqlpvc
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/presslabs/mysql-operator/pkg/options"
+	core "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName = "controller.mysqlpvc"
+	pvcFinalizer   = "pf9.io/MysqlPvcFinalizer"
+)
+
+var log = logf.Log.WithName(controllerName)
+
+// PvcReconciler manages lifetime of PVCs created by MysqlCluster with finalizers
+type PvcReconciler struct {
+	client.Client
+	scheme *runtime.Scheme
+	opt    *options.Options
+}
+
+func newPvcReconciler(mgr manager.Manager) *PvcReconciler {
+	return &PvcReconciler{Client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+		opt:    options.GetOptions(),
+	}
+}
+
+// Add creates a new Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+// USER ACTION REQUIRED: update cmd/manager/main.go to call this mysql.Add(mgr) to install this Controller
+func Add(mgr manager.Manager) error {
+	return add(mgr, newPvcReconciler(mgr))
+}
+
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to PVC
+	return c.Watch(&source.Kind{Type: &core.PersistentVolumeClaim{}}, &handler.EnqueueRequestForObject{})
```

Comment:
What do you think about watching only mysql-operator related PVCs, by using `source.Informer` (https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/source#Informer)? The informer would allow to watch only PVCs owned by a `MysqlCluster` and it also gives us a "free" resync as we need to loop over PVCs every n minutes to remove the finalizer and trigger the actual deletion.

For this you can pin controller-runtime dep to commit 5bccddbd327acf892546d58b8a117c19c007142b for the moment, until v0.1.5 gets released.